src/Controller.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace Magento\Upward;
+
+class Controller
+{
+    /** @var \Zend\Http\PhpEnvironment\Request */
+    private $request;
+
+    /** @var Context */
+    private $context;
+
+    /** @var Definition */
+    private $definition;
+
+    /** @var DefinitionTraverser */
+    private $definitionTraverser;
+
+    public function __construct(
+        \Zend\Http\PhpEnvironment\Request $request
+    ) {
+        $this->request = $request;
+        $this->context = Context::fromRequest($request);
+        // this value will need to be sourced from config?
```

Comment:
Or passed to the constructor, just like `$request` is.

`bin/upward` can get this path from environment variables and the pass it on to the class. I had not added a dotenv library but I think putting one in `require-dev` would make sense if we want to go this route.

Reply:
Sourcing config from environment variable now and passing via constructor.